Ginger/GingerCore/DataSource/ActDSTableElement.cs

```diff
@@ -112,6 +112,17 @@ public override void Execute()
                     else
                         outVal = ETERC.ValueCalculated;
                     break;
+                case eControlAction.DeleteRow:
+                    ValueExpression veDel = new ValueExpression(RunOnEnvironment, RunOnBusinessFlow, DSList);
```

Comment:
why below 3 lines repeat in almost all switch cases?
you can improve the code so it will be done before the switch case and then in switch case do the validations on it per case.
i'm pushing this code for release but please note to yourself to improve this code next time you working on it.

                    ValueExpression veDel = new ValueExpression(RunOnEnvironment, RunOnBusinessFlow, DSList);
                    veDel.Value = ValueExp;
                    outVal = veDel.ValueCalculated;